sycl/include/sycl/builtins.hpp

```diff
@@ -1377,6 +1377,198 @@ detail::common_rel_ret_t<T> signbit(T x) __NOEXC {
       __sycl_std::__invoke_SignBitSet<detail::internal_rel_ret_t<T>>(x));
 }
 
+// marray relational functions
+template <typename T,
+          typename = detail::enable_if_t<
+              detail::is_mgenfloat<T>::value &&
+                  detail::is_svgenfloat<detail::marray_element_type<T>>::value,
+              T>>
+sycl::marray<bool, detail::marray_size<T>> isequal(T x, T y) __NOEXC {
```

Comment:
Why can't we use `T::size()` here directly?

Reply:
This is just for code readability

Reply:
Not that I'm strongly opposed to this, but `T::size()` seems readable enough to me, IMO, this just adds more verbosity.

Reply:
Ok, changed to `T::size()`